video/updater.h

```diff
@@ -76,22 +78,23 @@ void VDUStreamProcessor::receiveFirmware() {
 
 		bytes_remain = readIntoBuffer(buffer, bytes_to_read);
 		if(bytes_remain) {
-			printFmt("Read buffer failed at byte %u!\n\r", remaining_bytes);
-			break;
+			printFmt("\n\rRead buffer failed at byte %u!\n\r", remaining_bytes);
+			return;
 		}
 
 		for(int i = 0; i < bytes_to_read; i++) {
 			code += ((uint8_t*)buffer)[i];
 		}
 
+		print(".");
+		remaining_bytes -= bytes_to_read;
+
 		err = esp_ota_write( update_handle, (const void *)buffer, bytes_to_read);
 		if (err != ESP_OK) {
-			printFmt("esp_ota_write failed, error=%d", err);
-			break;
+			printFmt("\n\resp_ota_write failed, error=%d\n'r", err);
+			discardBytes(remaining_bytes + 1);
+			return;
```

Comment:
this failure case and the one above both need to properly abort the update process...

there's a minimal chance (1 in 256) that a failure at this point could have led to an incomplete VDP being activated and attempted to be booted, which this resolves